### PR TITLE
[Adjustments] Update line item taxes

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Updates the way line item included taxes are fetched, so we no longer depend on the `included_tax` column in the `spree_adjustments` table. This column is basically deprecated and in the process of being removed.

The same figure (included tax amount on a line item) is available with: `line_item.adjustments.tax.inclusive.sum(:amount)`.

#### What should we test?

Line Item tax amounts are displayed correctly when using included taxes.

#### Release notes

Changelog Category: Technical changes

### CODE REVIEW

File: app/services/tax_rate_finder.rb [hunk 1]
```diff
@@ -75,7 +75,7 @@ def line_item_tax_category(enterprise_fee, line_item)
   # to the included tax.
   def find_closest_tax_rates_from_included_tax(amount, included_tax)
     approximation = (included_tax / (amount - included_tax))
-    return [] if approximation.infinite? || approximation.zero?
+    return [] if approximation.infinite? || approximation.zero? || approximation.nan?
```
Comment (andrewpbrett) [inline app/services/tax_rate_finder.rb:78]: 😂 `NaN` strikes again

**Review by andrewpbrett — APPROVED** [review]
This looks pretty straightforward and like a more sensible approach if I'm understanding it all correctly. 👍

**Review by sauloperez — APPROVED** [review]

**Comment by Matt-Yorkley** [comment]
Rebased :+1:

**Comment by filipefurtad0** [comment]
Hey @Matt-Yorkley,

This seems like a relevant and recurring failing spec:

`rspec ./spec/services/paypal_items_builder_spec.rb:59 # PaypalItemsBuilder listing adjustments should add up to the order total, minus any additional tax and the shipping cost
`

It's expecting `items_total` to be a negative value... Maybe something on how `order_tax_total` is defined?

**Comment by Matt-Yorkley** [comment]
I think that's fixed already in master, maybe it just needs a rebase :+1:

**Comment by filipefurtad0** [comment]
Hi @Matt-Yorkley ,

Checked GST (inclusive tax) adjustments, operating on line items.

- Previously placed orders remain correct (regression test)
- New orders render the correct tax value
- Changes on the tax rate (superadmin) have effect on new orders only
- Taxes on enterprise fees work as before (regression test)
- Items with price zero do not contribute to taxes; only their associated enterprise fees

As an example, the pic below:

image


Awesome Matt, we're good to go here :+1:

**Pull request merged**